src/FormField/FormField.tsx

```diff
@@ -25,47 +25,121 @@ export namespace FormField {
     style?: React.CSSProperties;
     /** an optional id to pass to top level element of the component */
     id?: string;
+    /** an optional hint describing what's the expected value for the field (e.g. sample value or short description) */
+    hint?:
+      | {
+          content: JSX.Element | string;
```

Comment:
I've recently find myself using `NonNullable<React.ReactNode>`. What do you think?